utils/zerovec/src/map2d/databake.rs

```diff
@@ -42,3 +42,119 @@ where
         quote! { unsafe { #[allow(unused_unsafe)] ::zerovec::maps::ZeroMap2dBorrowed::from_parts_unchecked(#keys0, #joiner, #keys1, #values) } }
     }
 }
+
+#[test]
+fn test_baked_map() {
+    test_bake!(
+        ZeroMap2d<str, str, str>,
+        const: unsafe {
+            #[allow(unused_unsafe)]
+            crate::ZeroMap2d::from_parts_unchecked(
+                unsafe {
+                    crate::VarZeroVec::from_bytes_unchecked(&[
+                        97u8, 114u8, 99u8, 97u8, 122u8, 0u8, 99u8, 117u8, 0u8, 101u8, 110u8, 0u8,
```

Comment:
issue: these tests are completely unsound (there is no length metadata), but they pass because VZV databake is a straight passthrough. Do you recall how you generated this data?

Discovered whilst investigating https://github.com/unicode-org/icu4x/issues/3510

Reply:
Wdym no length metadata?

Reply:
Inside the VZV? I might have just copied the byte slices from other tests. You can use some testdata VZV here.

Reply:
Gotcha, will do.

When I say no length metadata, I'm saying that this is just a string of ASCII bits, without any information on where the boundaries are. The first ascii bits get treated as lengths and this indexes out of bounds.